### DISCUSSION TITLE
Constant Function Market Makers，about The trade function

### DISCUSSION THREAD

**Discussion by muzzby** [Jeiwan/uniswapv3-book#61] [Milestone 0. Introduction]
I read about constant function market makers and saw a formula in the 'Trading function', (x+rΔx)(y−Δy)=k. In point 5 of the explanation, it says 'the quantity of token0 in the pool has changed (x+rΔx), and the quantity of token1 has also changed (y-Δy)'. I have a doubt that shouldn't the change in quantity of token0 be (x+rΔx+Δx)? Please advise, thank you!

**Comment by Jeiwan** [comment]
The formula is correct. The initial reserves are `x` and `y`, and swapping increases `x` and decreases `y` (or the other way around). Thus, $\Delta x$ is added and $\Delta y$ is removed. Since fees in Uniswap are taken from input amounts (the token being sold), `r` is applied to $\Delta x$. `r` in the formula is $1 - \text{fee}$, where `fee` is 0.003 (Uniswap V3 has different fee tiers). So, $r \Delta x$ is $0.997 * \Delta x$.

**Reply by bluenights004** [reply]
agree with muzzby, it seems + ∆x  is missing in formula which represents adding the increase cause by swap, the liquidity provider fee (rΔx) is the only item added in previous reserve x ?

in other words, previous reserve (x) + liquidity provider fee (r∆x) + increase in x (∆x) due to swap = new reserve balance of x, should be included in the formula?

**Reply by Brivan-26** [reply]
@bluenights004 You're wrong about something: 
> in other words, previous reserve (x) + liquidity provider fee (r∆x) + increase in x (∆x) due to swap

(r∆x) is not the liquidity provider fee. The liquidity provider fee is constant, that is taken from the sold token. 
If you are about to sell ∆x tokens0, the pool takes `fee` from it, so the real amount of token 0 being added to the pool is `(1- fee)*∆x`.
In the context of Uniswap, consider the following example: We have a pool of token0/token1. Initially, we have `100` of tokens each.
 supposing you are selling `40` of token0 to get `x` amount of token1. After the trade happens, the pool will not have `100 + 40` of token0 as the protocol takes seme fees, in uniswap it is 0.3%. So, `0.3%` of `40` token0 you're selling will be reducted, which means the pool will have `100 + 0.97*40` of token0.
 I hope this clears the confusion
 cc @muzzby
